### PR TITLE
[MODULAR] Adds height scaling

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Title. 

You can now set your characters height using TG's height scaling. Works with size scaling.

## Why It's Good For The Game

Size scaling is ass. Its just fucking ass. Most people who use it want it as a representation of their charactyers height and have to ENTIRELY SHRINK or GROW. It SUCKS. I want to play a tall or short character that isnt SHRUNK or INFLATED.

## Proof Of Testing

image

Humans, left to right: 0.8 size scaling, shortest scaling. Shortest, short, medium, tall, taller, tallest.
Teshari, left to right: Shortest, Tallest.
The taur is a 1.2 size scaling with oversized and tallest, the absolute extreme. The weird artifating on the bottom happens on most taurs, and is avoidable by not selecting tallest.

## Changelog

:cl:
add: You can now set your characters height
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
yes the tesh have weird legs, my tesh has cybernetic legs

**Comment by nikothedude** [comment]
I FORGOT

**Comment by ReturnToZender** [comment]
Love this!!

**Comment by AllCorruptingWyrm** [comment]
THANK GOD

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/bubbers/height_scaling.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { FeatureChoiced, FeatureDropdownInput } from '../../base';
+
+export const height_scaling: FeatureChoiced = {
+  name: 'Height',
```
Comment (Majkl-J) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/bubbers/height_scaling.tsx:4]:
```suggestion
  name: 'Body Height',
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
So it's together with the other pref

**Review by Majkl-J — COMMENTED** [review]
This is cool and all but
image
image

Yea the performance on these chokes a lot

**Comment by nikothedude** [comment]
> This is cool and all but image image
> 
> Yea the performance on these chokes a lot

is this from live?

**Comment by Majkl-J** [comment]
> is this from live?

Yup freshly profiled

**Comment by nikothedude** [comment]
its worth noting that dwarfs and spacer both use the same exact framework as this

**Comment by nikothedude** [comment]
ill look into this to see if it can be optimized but if not we'll have to consider if this is worth the cost

**Review by Majkl-J — APPROVED** [review]
Optimizations actually made this run so fast its impact is barely noticeable.

Anyhow yes approved I'll look to merging this after like a week, needs some testing period

**Comment by HissyNays** [comment]
This is a pretty good attempt. However I feel like some of my characters got messed up and I don't know how to fix them. 
What needs to be actually made is two sliders, one for height, one for width of your character's sprite. That way it will be the most customizable and best. I've seen a different server do it and it was amazing.

**Comment by Arturlang** [comment]
What happens if you have both the spacer quirk and increase your height?

**Comment by IgiariValkyr** [comment]
Cuts off the top of your sprite if it goes over the 32 pixel limit.
dreamseeker_2024-03-18_03-51-31
dreamseeker_2024-03-18_03-51-02

**Comment by nikothedude** [comment]
> Cuts off the top of your sprite if it goes over the 32 pixel limit. dreamseeker_2024-03-18_03-51-31 dreamseeker_2024-03-18_03-51-02

Not sure how to fix this but I doubt I can, itd need to be a TG fix

**Comment by ReturnToZender** [comment]
As a tall dude that doesn't take up a whole tile I am winning

**Comment by Iamgoofball** [comment]
this causes issues for client-side performance, filters aren't cheap, please consult @MrMelbert before proceeding

**Comment by Majkl-J** [comment]
> this causes issues for client-side performance, filters aren't cheap, please consult @MrMelbert before proceeding

Already did
https://github.com/Bubberstation/Bubberstation/pull/1264/commits/caab4e479a1c90f5f8c53e8c8cc8d773573cad88

**Comment by Iamgoofball** [comment]
did you read the code you linked before replying to me

**Comment by Majkl-J** [comment]
> did you read the code you linked before replying to me

This is specifically what melbert has given to me and niko when we asked him about the problem. So yes

**Comment by Iamgoofball** [comment]
if you had read the words i wrote you would see i am talking about clientside performance, not serverside performance

**Comment by Majkl-J** [comment]
> if you had read the words i wrote you would see i am talking about clientside performance, not serverside performance

This is all we were given. Stuff has been testmerged for a while now with no issues. I dunno what to tell you

**Comment by IgiariValkyr** [comment]
> > Cuts off the top of your sprite if it goes over the 32 pixel limit. dreamseeker_2024-03-18_03-51-31 dreamseeker_2024-03-18_03-51-02
> 
> Not sure how to fix this but I doubt I can, itd need to be a TG fix

This issue was never fixed.

**Comment by nikothedude** [comment]
I know

**Pull request merged**